crates/net/network/src/cache.rs

```diff
@@ -36,14 +44,33 @@ impl<T: Hash + Eq> LruCache<T> {
         false
     }
 
+    /// Same as [`Self::insert`] but returns a tuple, where the second index is the evicted value,
+    /// if one was evicted.
+    pub fn _insert_with_eviction_feedback(&mut self, entry: T) -> (bool, Option<T>) {
+        if self.inner.insert(entry) {
+            if self.limit.get() == self.inner.len() {
+                // remove the oldest element in the set
+                return (true, self.remove_lru())
+            }
+            return (true, None)
+        }
+        (false, None)
+    }
+
     /// Remove the least recently used entry and return it.
     ///
-    /// If the `LruCache` is empty this will return None.
+    /// If the `LruCache` is empty or if the eviction feedback is
+    /// configured, this will return None.
     #[inline]
     fn remove_lru(&mut self) -> Option<T> {
         self.inner.pop_front()
     }
 
+    /// Expels the given value. Returns true if the value existed.
+    pub fn _remove(&mut self, value: &T) -> bool {
```

Comment:
I think we can drop the _ prefix here, no need to hide it

Reply:
tell that to clippy, jk